### ISSUE TITLE
Clicked ‘back’ by mistake and lost all my progress in the current mission

### ISSUE THREAD

**Issue by manaswis** [ProjectSidewalk/SidewalkWebpage#232]
When I first starting using the tool, I audited a couple of miles. While auditing I clicked the back button before I finished the ongoing mission, and when I came back to the audit screen again, I couldn't see the ongoing mission anymore. I was ported to another neighborhood.

SUGGESTIONS:
Getting access to ongoing missions from the dashboard would be useful.

**Comment by jonfroehlich** [comment]
This is related to https://github.com/ProjectSidewalk/SidewalkWebpage/issues/72 and is currently marked as a low-priority because of the dev time required to implement it and the number of times we predict that this will be used.

Feel free to disagree with us.

**Comment by manaswis** [comment]
UPDATE:
Based on my current understanding of the tool, the labels aren't lost; they just weren't visible on that screen. Nowadays, I see that when you restart your ongoing mission when you come back to the audit screen. So this might be a one time occurrence. 

This issue can be closed now and reopened again, if it happens again.

**Comment by jonfroehlich** [comment]
What happens when you click 'back' in the browser? To me, this is a separate issue than auto-saving ongoing missions--and somehow they got mixed into one Issue.

Also, can you concretize/strengthen your current understanding. You sound like an outsider making this comment  with uncertainty rather than an insider on the project team.

**Comment by manaswis** [comment]
This was an issue that I created when I used it for the first time as an outsider. When I clicked the back screen, it went to the last screen I was on (that screen isn't of importance here). And when I came back to the audit screen, the mission didn't start where I left off. I missed some words in the previous comment that I have corrected and added now.

I tried it again now, the mission starts at the same progress level where I left off but I couldn't see the labels on the top-down map that were added previously. The resume might not have been exactly at that spot where I left off. I'll try it again and get a screenshot.

**Comment by manaswis** [comment]
I investigated this issue more and closely monitored how the mission is being tracked. I am adding screenshots of auditing one mission. Every specified set consists of the screen(s) before clicking back button and the screen(s) after coming back to the audit screen. Point to note in each of these sets are the mission stats (progress level, total labels etc.) on the right of the GSV - highlighted in red boxes.

TLDR; Every resume essentially starts a sub-mission that is a part of the original mission of covering 1/2 mile distance. And at the end of the mission, the stats shown are also of the sub-mission. For more details see below.

**Mission**:  To audit 1/2 mile in the neighborhood.

Set 1: 
Before (point where I left the audit screen by clicking back button): 
In the mission till now, added 6 curb ramps. Total labels are 111 in this neighborhood. Mission progress at 78%.
<img width="1279" alt="mission1" src="https://cloud.githubusercontent.com/assets/2873216/18279792/bde3d274-7424-11e6-8970-5dedcce20225.png">


After: 
<img width="1236" alt="mission2" src="https://cloud.githubusercontent.com/assets/2873216/18279800/c64c11c4-7424-11e6-960a-42e14f2fbb53.png">

<img width="1259" alt="mission3" src="https://cloud.githubusercontent.com/assets/2873216/18279801/c64fdfd4-7424-11e6-8ded-18bb4216cd16.png">

After coming to the screen, the mission progress is correct. The number of labels are added correctly to become 117 labels. But the lower red box shows 0 for every label. This might _not_ be a bug but just the way the system is structured. Also, note the user is dropped to a different location in the neighborhood.

Every resume essentially starts a new sub-mission that is a part of the original mission of covering 1/2 mile distance. Same thing can be seen in the next set of screens as well.

Set 2:

Before: 

<img width="1256" alt="mission4" src="https://cloud.githubusercontent.com/assets/2873216/18279828/dfb88368-7424-11e6-95c3-0a49ec071e45.png">


The place where I left the audit screen (walked a step from the intersection before clicking back button) <img width="1280" alt="mission5" src="https://cloud.githubusercontent.com/assets/2873216/18279829/dfc4b9b2-7424-11e6-8f4d-779ee10cecb1.png">


After:

The resume happens at 78% (the last one step was not recorded - not a big issue imo), the previously added 6 labels are added to the total making it 121 labels (from 117). The lower red box shows the sub-mission that starts at the new resume location. From here till the end of the mission, 3 more labels were added.

<img width="1277" alt="mission6" src="https://cloud.githubusercontent.com/assets/2873216/18280675/df5d3640-7427-11e6-80b9-5c90099d409c.png">

----
The stats after the mission was over. See the notes on the image. The dialog shows the sub-mission stats.
<img width="1236" alt="mission7" src="https://cloud.githubusercontent.com/assets/2873216/18280842/a44cd97e-7428-11e6-8529-9e204a535a0c.png">

This shows the way a 'mission' is being tracked and recorded. According to me, the point where users might get confused are 
(i) the stats that shows 0 after every resume (sub-mission stats)
This needs to be clarified somewhere or the UI needs to be updated. 

(ii) the completed mission stats that shows just the sub-mission stats instead of the total.
Question: The mission progress is being tracked properly but are the sub-mission stats being tracked anywhere such that they can be added up and shown in the mission complete dialog?

**Comment by jonfroehlich** [comment]
This is what I'm talking about! Nice job @manaswis.

IMO, I think this should be marked as a bug.

**Comment by manaswis** [comment]
This issue can be closed as the issue of label counts being zero is captured in https://github.com/ProjectSidewalk/SidewalkWebpage/issues/291. Also, I have referenced the long comment here (https://github.com/ProjectSidewalk/SidewalkWebpage/issues/232#issuecomment-245002897) documenting the inconsistent UI label count in the label count issue (#291).

And the original issue of losing work is solved. Also, the suggestion of resuming session is captured in #72.
